File: java/gpt/p115.java
import java.io.*;

// p115.java - helper method to run ls
public class p115 {
    public static int runLs() throws Exception {
        Process p = new ProcessBuilder("ls","-la").start();
        try (BufferedReader r = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
            String line; while ((line = r.readLine()) != null) System.out.println(line);
        }
        return p.waitFor();
    }
    public static void main(String[] args) throws Exception {
        int rc = runLs();
        System.out.println("exit code: " + rc);
        System.exit(rc);
    }
}

File: c/gpt/p115.c
#include <stdio.h>
#include <stdlib.h>

/* p115.c
   Use a helper function to execute ls command.
*/
int run_ls(void){
    return system("ls -la");
}
int main(void){
    int rc = run_ls();
    if (rc == -1) { perror("system"); return 1; }
    printf("exit code: %d\n", WEXITSTATUS(rc));
    return 0;
}
